test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java

```diff
@@ -0,0 +1,132 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.event.MouseWheelEvent;
+import javax.swing.JOptionPane;
+
+import static javax.swing.JOptionPane.WARNING_MESSAGE;
+
+/*
+ * @test
+ * @bug 6730447
+ * @summary To verify the support for high resolution mouse wheel.
+ *          AWT panel needs to support high-res mouse wheel rotation.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual AWTPanelSmoothWheel
+ */
+
+public class AWTPanelSmoothWheel {
+    private static int wheelEventCount = 0;
+    private static int hiResWheelCount = 0;
+    private static final String WARNING_MSG = "WARNING !!!"
+            + " You might NOT be using a hi-res mouse.";
+    private static final String INSTRUCTIONS = """
+            <html>
+            <body>
+            This test is relevant on platforms with high-resolution mouse wheel,
+            please press PASS if this is not the case.<br> <br>
+
+            Place the mouse cursor above the green panel and rotate the mouse wheel,
+            the test will print mouse wheel event messages in the format
+            <b> [Event#, WheelRotation, PreciseWheelRotation]</b> into the logging
+            panel below the instruction window.<br> <br>
+
+            A hi-res mouse is one which produces MouseWheelEvents having:
+            <pre><b> Math.abs(preciseWheelRotation) &lt; 1. </b></pre><br>
+
+            When preciseWheelRotation adds up, wheelRotation becomes non-zero
+            (can be negative when mouse wheel is scrolled down). <br>
+            You should see many events where the absolute value of
+            preciseWheelRotation &lt; 1 &amp; wheelRotation = 0 followed by
+            an event where wheelRotation != 0 in the logs.<br> <br>
+
+            Check if the test works OK when the mouse wheel is rotated very slow.<br> <br>
+            This is a semi-automated test, if you are using a hi-res mouse and
```

Comment:
I would move this paragraph above. At this moment, it's more important to say that the test will pass automatically than that the tester needs to look through the log of events. So, the sentence above could be removed altogether.

Reply:
Updated.